cunumeric/deferred.py

```diff
@@ -442,26 +657,32 @@ def set_item(self, key, rhs):
         # Check to see if this is advanced indexing or not
         if is_advanced_indexing(key):
             # Create the indexing array
-            index_array = self._create_indexing_array(key)
-            if index_array.shape != rhs.shape:
-                raise ValueError(
-                    "Advanced indexing array does not match source shape"
-                )
-            if self.ndim != index_array.ndim:
-                raise NotImplementedError(
-                    "need support for indirect partitioning"
-                )
+            (
+                copy_needed,
+                lhs,
+                index_array,
+                self,
+            ) = self._create_indexing_array(key, True)
+            rhs = self.runtime.to_deferred_array(rhs)
```

Comment:
I think `rhs` has already been converted to a `DeferredArray` through `@auto_convert([2])`.

Reply:
Yes, you are right